src/datasources/alarms/query-type-handlers/alarms-trend/AlarmsTrendQueryHandler.ts

```diff
@@ -1,17 +1,115 @@
 import { DataFrameDTO, DataQueryRequest, FieldType } from '@grafana/data';
 import { AlarmsQueryHandlerCore } from 'datasources/alarms/query-type-handlers/AlarmsQueryHandlerCore';
 import { defaultAlarmsTrendQuery } from 'datasources/alarms/constants/DefaultQueries.constants';
-import { AlarmsTrendQuery } from 'datasources/alarms/types/AlarmsTrend.types';
+import { AlarmsTrendQuery, AlarmTransitionEvent, AlarmWithNumericTimeInTransitions } from 'datasources/alarms/types/AlarmsTrend.types';
+import { Alarm, AlarmTransitionType, TransitionInclusionOption } from 'datasources/alarms/types/types';
 
 export class AlarmsTrendQueryHandler extends AlarmsQueryHandlerCore {
   public readonly defaultQuery = defaultAlarmsTrendQuery;
 
   public async runQuery(query: AlarmsTrendQuery, options: DataQueryRequest): Promise<DataFrameDTO> {
-    // TODO: alarms trend query logic will be implemented here in future
+    const { start, end, intervalMs } = this.extractTimeParameters(options);
+    const filter = this.getTrendQueryFilter(query, start, end);
+    const alarms = await this.queryAlarmsUntilComplete({ filter, transitionInclusionOption: TransitionInclusionOption.All });
+    const alarmsWithTimestamp = this.enrichTransitionsWithTimestamp(alarms);
+    const trendData = this.countActiveAlarmsPerInterval(alarmsWithTimestamp, start, end, intervalMs);
+
     return {
       refId: query.refId,
       name: query.refId,
```

Comment:
```suggestion
      name: query.refId
```

For Alarms Count query type, we return like this. Can we follow the same here?